Fix auto-complete not suggesting custom class names on identifiers

poohcom1 opened the pull request:
Fixes #69941

In 4.x, auto-complete will suggest custom classes only when completing types, not identifiers. This makes it auto-complete for identifiers as well, as this was the behavior in 3.x.

![image](https://user-images.githubusercontent.com/74857873/207100492-c069e364-327b-44d3-beeb-881c9d58bf40.png)

![image](https://user-images.githubusercontent.com/74857873/207100451-c0cfb131-e069-48e0-9fdd-313fc33a18d5.png)

adamscott commented:
CC @anvilfolk

adamscott commented on modules/gdscript/gdscript_editor.cpp:
```diff
@@ -1271,6 +1271,14 @@ static void _find_identifiers(const GDScriptParser::CompletionContext &p_context
 		r_result.insert(option.display, option);
 	}
+
+	// Global scripts
```
I would change this comment to `// Global classes`, but it's a small nitpick.

poohcom1 replied:
Yeah I agree; I'll update it now

adamscott reviewed (approved):
LGTM.

poohcom1 committed:
Add identifier completion for custom classes.
Previously, custom class would only auto-complete for types in GDScript.
This applies it to identifiers as well.

vnen commented:
Thanks!
